### PR TITLE
MRG: Refactor evoked and tfr image plot mask to share code

### LINKED ISSUE #5061: Refactor evoked image and tfr image plot masking code
`evoked.plot_image(..., mask=...)` and the corresponding TFR method should share code for masking.

@cbrnr

### PR DESCRIPTION
**Pull request by jona-sassenhagen**
Closes #5061

### CODE REVIEW

File: mne/viz/utils.py [hunk 1]
```diff
@@ -2510,3 +2510,121 @@ def _set_title_multiple_electrodes(title, combine, ch_names, max_chans=6,
...
+                           antialiased=False, levels=[.5])
+        time_lims = times[[0, -1]]
+        ylim = yvals[[0, -1]]
+
+    ax.set_xlim(time_lims[0], time_lims[-1])
+    ax.set_ylim(ylim)
+
+    if (draw_mask or draw_contour) and mask is not None:
+        if mask.all():
+            t_end = ", all points masked)"
```
Comment (mmagnuski) [inline mne/viz/utils.py]: does it need a special case? :)

Comment (jona-sassenhagen) [reply mne/viz/utils.py]: It doesn't *need* it ...

Comment (jona-sassenhagen) [reply mne/viz/utils.py]: Actually, yes, I think it's quite helpful - it unambiguously tells you nothing is significant.

Comment (mmagnuski) [reply mne/viz/utils.py]: > nothing is significant

:( 
Consider "nothing is significant but here is a funny gif to cheer you up: [link]" ;)

Comment (jona-sassenhagen) [reply mne/viz/utils.py]: Haha, yes.

**Comment by jona-sassenhagen** [comment]
A few examples so far:

unknown-35_img

unknown-34_im

**Comment by agramfort** [comment]
needs rebase

**Comment by jona-sassenhagen** [comment]
Yeah, this is rather WIP ... I think I'll return to it after the release?

**Comment by larsoner** [comment]
@jona-sassenhagen let me know when you want a round of review (same goes for your refactoring PR)

**Comment by agramfort** [comment]
this does not affect at all any of our plots?

**Comment by jona-sassenhagen** [comment]
Hopefully not - it should be all backend work.

It also partially affects text labels for some corner cases though. I'll trigger the CIs.

**Comment by jona-sassenhagen** [comment]
https://8015-1301584-gh.circle-artifacts.com/0/html/auto_tutorials/plot_stats_cluster_erp.html

https://8015-1301584-gh.circle-artifacts.com/0/html/auto_examples/stats/plot_sensor_regression.html

https://8015-1301584-gh.circle-artifacts.com/0/html/auto_examples/time_frequency/plot_time_frequency_erds.html

Here are the plots in question. Looking good!

**Comment by agramfort** [comment]
good to go then?

**Comment by jona-sassenhagen** [comment]
Yes!

@larsoner consider the last plot:

https://8015-1301584-gh.circle-artifacts.com/0/html/auto_examples/stats/plot_sensor_regression.html

Awkward, right?

(I should also change the `Nave` of linear_regression output to match n_epochs.)

**Comment by larsoner** [comment]
Yes :(

Maybe `ha='right'` to the axis boundary? If that doesn't work, maybe left-align to the right axis boundary? And maybe `va='top'` on the top of the axis boundary? Feel free to play with it if you have time.

**Comment by jona-sassenhagen** [comment]
@larsoner that's a new one for me ..?

`OSError: [Errno 28] No space left on device`

Should I just do it again? I'd like to trigger a few.

**Comment by larsoner** [comment]
You can tell it to rebuild on the CircleCI website instead of pushing a
commit (this will save Travis and AppVeyor some time)

**Comment by jona-sassenhagen** [comment]
Ah I see.

**Comment by jona-sassenhagen** [comment]
Circle looks good. Feel free to merge.

The Nave stuff I'm not looking at here ...

**Comment by larsoner** [comment]
I should be able to check the changed examples tomorrow. Should I just look at the two you touched?

**Comment by jona-sassenhagen** [comment]
They should look the same as before.

Main ones would be

auto_examples/time_frequency/plot_time_frequency_erds.html

auto_tutorials/plot_stats_cluster_erp.html

auto_examples/stats/plot_sensor_regression.html

**Comment by larsoner** [comment]
plot_time_frequency_erds.html is not identical:

https://martinos.org/mne/dev/auto_examples/time_frequency/plot_time_frequency_erds.html#sphx-glr-auto-examples-time-frequency-plot-time-frequency-erds-py

https://8034-1301584-gh.circle-artifacts.com/0/html/auto_examples/time_frequency/plot_time_frequency_erds.html

But this seems like an improvement.

The percentage is inverted here:

https://martinos.org/mne/dev/auto_examples/stats/plot_sensor_regression.html

https://8034-1301584-gh.circle-artifacts.com/0/html/auto_examples/stats/plot_sensor_regression.html

**Comment by jona-sassenhagen** [comment]
I'm not sure what would be correct ... But I guess if 24% of points are visible, it should say 76 are masked right ..?

**Comment by larsoner** [comment]
Yeah

**Comment by agramfort** [comment]
agreed

**Comment by jona-sassenhagen** [comment]
Anyone else?

**Comment by cbrnr** [comment]
I don't have time for a code review, but my ERDS example looks good. Although I'd prefer no contour - where exactly are these arguments documented?

**Comment by jona-sassenhagen** [comment]
Ah right, API oversight - tfr.plot doesn't have `mask_style` as an arg. On it.

**Comment by jona-sassenhagen** [comment]
I don't believe the coverage hit is real ... So this would be green. @cbrnr , any further comments?

**Comment by cbrnr** [comment]
LGTM!

**Comment by agramfort** [comment]
thx @jona-sassenhagen

**Comment by jona-sassenhagen** [comment]
Great, thanks!

**Pull request merged**